clients/drcachesim/tools/view.cpp

```diff
@@ -591,6 +591,11 @@ view_t::parallel_shard_memref(void *shard_data, const memref_t &memref)
         return true;
     }
 
+    // XXX: We could potentially use instr_decode_cache_t here (i#7113) and avoid the
+    // repeated instr decoding logic. However, we want to preserve the legacy view
+    // tool output format which uses disassemble_to_buffer, and disassemble_to_buffer
+    // does the decoding on its own internally, while adding a bunch of non-trival
+    // stuff to the output string.
```

Comment:
Add that the decode cache cannot currently replace the view decoding because it does not set the ISA mode for regdeps.

Reply:
Moved the ISA mode setting to init()